test/game/ShipTest.java

```diff
@@ -145,58 +167,133 @@ void rechargeWeaponesOperated() {
 
     }
 
+    @Test
+    void calculateOxygenState_emptyTank() {
+        ship.setOxygenLevel(0);
+        ship.calculateState();
+        assertTrue(ship.getOxygenLevel() > 0);
+    }
 
+    @Test
+    void calculateOxygenState_almostFullTank() {
+        ship.setOxygenLevel(99);
+        ship.calculateState();
+        assertEquals(100, ship.getOxygenLevel());
+    }
+
+    @Test
+    void calculateOxygenState_fullTank() {
+        ship.calculateState();
+        assertEquals(100, ship.getOxygenLevel());
+
+    }
+
+
+    @Test
+    void calculateShieldState_emptyShield() {
+        ship.setShield(0);
+        ship.calculateState();
+        assertTrue(ship.getShield() > 0);
+    }
 
     @Test
-    void calculateState() {
-//        TODO: Implement test
+    void calculateShieldState_almostFullShield() {
+        ship.setShield(0.99);
+        ship.calculateState();
+        assertEquals(1, ship.getShield());
     }
 
     @Test
-    void addCrewmember() {
+    void calculateShieldState_fullShield() {
+        ship.calculateState();
+        assertEquals(1, ship.getShield());
+    }
+
+    @Test
+    void calculateShieldState_operatedBonus()
+    {
+        ship.setShield(0.14);
+        ship.calculateState();
+        double notOperated = ship.getShield();
+
+
+        ship.addCrewmember(new Person(Ship.getShieldId()));
 
+        ship.setShield(0.14);
+        ship.calculateState();
+        double operated = ship.getShield();
 
+        assertTrue(operated > notOperated);
+    }
+
+    @Test
+    void calculateCrewState_crewSuffocating()
+    {
+        Person p = new Person(1);
+        ship.addCrewmember(p);
+        ship.setOxygenLevel(0);
+        ship.calculateState();
 
-        assertEquals(0, ship.getCrewCount());
+        assertTrue(p.getHealthPoints() < 100);
+    }
 
+    @Test
+    void addCrewmember_addSinglePerson() {
         ship.addCrewmember(new Person(3));
 
         assertEquals(1, ship.getCrewCount());
         ship.addCrewmember(new Person(4));
         ship.addCrewmember(new Person(1));
         assertEquals(3, ship.getCrewCount());
+    }
 
+    @Test
+    void addCrewmember_addMultiplePeople() {
```

Comment:
Testy `_addSinglePerson` i `_addMultiplePeople` sprawdzają w sumie chyba to samo, nie? Można zostawić tylko jeden.